File: TheaterSuggestions/CSharp/SeatingPlaceSuggestions/SuggestionIsMade.cs
namespace SeatsSuggestions;

/// <summary>
///     Occurs when a Suggestion is made.
/// </summary>
public class SuggestionIsMade(SeatingOptionIsSuggested seatingOptionIsSuggested)
{
    private readonly List<SeatingPlace> _suggestedSeats = seatingOptionIsSuggested.Seats;

    public int PartyRequested { get; } = seatingOptionIsSuggested.PartyRequested;
    public PricingCategory PricingCategory { get; } = seatingOptionIsSuggested.PricingCategory;

    public IReadOnlyList<SeatingPlace> SuggestedSeats => _suggestedSeats;

    public IEnumerable<string> SeatNames()
    {
        return _suggestedSeats.Select(s => s.ToString());
    }

    public bool MatchExpectation()
    {
        return _suggestedSeats.Count == PartyRequested;
    }
}

File: TheaterSuggestions/Java/ScreeningSeatingArrangementSuggestions/src/main/java/org/weaveit/seatingplacesuggestions/SuggestionIsMade.java
package org.weaveit.seatingplacesuggestions;

import java.util.List;

public class SuggestionIsMade {

    private final List<SeatingPlace> suggestedSeats;
    private final int partyRequested;
    private final PricingCategory pricingCategory;

    public SuggestionIsMade(SeatingOptionIsSuggested seatingOptionIsSuggested) {
        this.suggestedSeats = seatingOptionIsSuggested.seats();
        this.partyRequested = seatingOptionIsSuggested.partyRequested();
        this.pricingCategory = seatingOptionIsSuggested.pricingCategory();
    }

    public List<String> seatNames() {
        return suggestedSeats.stream().map(SeatingPlace::toString).toList();
    }

    public boolean matchExpectation() {
        return suggestedSeats.size() == partyRequested;
    }

    public PricingCategory pricingCategory() {
        return pricingCategory;
    }
}
